crates/rooch-key/src/keystore/account_keystore.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::key_derive::{
+    derive_address_from_private_key, derive_private_key_from_path, encrypt_key,
+    generate_derivation_path, generate_new_key_pair, hash_password,
+};
+use crate::keystore::ImportedMnemonic;
+use bip32::DerivationPath;
+use bip39::{Language, Mnemonic, Seed};
+use fastcrypto::encoding::{Base64, Encoding};
+use rooch_types::key_struct::{MnemonicData, MnemonicResult};
+use rooch_types::{
+    address::RoochAddress,
+    authentication_key::AuthenticationKey,
+    crypto::{PublicKey, RoochKeyPair, Signature},
+    error::RoochError,
+    key_struct::{EncryptionData, GeneratedKeyPair},
+    transaction::rooch::{RoochTransaction, RoochTransactionData},
+};
+use serde::Serialize;
+
+pub trait AccountKeystore {
+    fn add_address_encryption_data(
+        &mut self,
+        address: RoochAddress,
+        encryption: EncryptionData,
+    ) -> Result<(), anyhow::Error>;
+    fn get_address_public_keys(
+        &self,
+        password: Option<String>,
+    ) -> Result<Vec<(RoochAddress, PublicKey)>, anyhow::Error>;
+    fn get_public_key(&self, password: Option<String>) -> Result<PublicKey, anyhow::Error>;
+    fn get_key_pairs(
+        &self,
+        address: &RoochAddress,
+        password: Option<String>,
+    ) -> Result<Vec<RoochKeyPair>, anyhow::Error>;
+    fn get_key_pair_with_password(
+        &self,
+        address: &RoochAddress,
+        password: Option<String>,
+    ) -> Result<RoochKeyPair, anyhow::Error>;
+
+    fn get_password_hash(&self) -> String;
+
+    fn get_if_password_is_empty(&self) -> bool;
+
+    fn set_password_hash_with_indicator(
+        &mut self,
+        password_hash: String,
+        is_password_empty: bool,
+    ) -> Result<(), anyhow::Error>;
+
+    fn update_address_encryption_data(
+        &mut self,
+        address: &RoochAddress,
+        encryption: EncryptionData,
+    ) -> Result<(), anyhow::Error>;
+    fn nullify(&mut self, address: &RoochAddress) -> Result<(), anyhow::Error>;
+
+    fn sign_hashed(
+        &self,
+        address: &RoochAddress,
+        msg: &[u8],
+        password: Option<String>,
+    ) -> Result<Signature, anyhow::Error>;
+
+    fn sign_transaction(
+        &self,
+        address: &RoochAddress,
+        msg: RoochTransactionData,
+        password: Option<String>,
+    ) -> Result<RoochTransaction, anyhow::Error>;
+
+    fn sign_secure<T>(
+        &self,
+        address: &RoochAddress,
+        msg: &T,
+        password: Option<String>,
+    ) -> Result<Signature, anyhow::Error>
+    where
+        T: Serialize;
+
+    fn addresses(&self) -> Vec<RoochAddress>;
+
+    fn generate_and_add_new_key(
+        &mut self,
+        mnemonic_phrase: Option<String>,
```

Comment:
Do we need this argument? The mnemonic shoud be a required field when the keystore create.

Reply:
needed when `rooch init` with `mnemonic phrase` option

Reply:
Maybe the init command should create a new keystore with mnemonic, we can refactor it later.